File: frontend/src/types/soil.ts
export interface SoilHealthData {
  health_score: number;
  degradation_risk: 'LOW' | 'MODERATE' | 'HIGH' | 'CRITICAL';
  organic_matter_percent: number;
  ph: number;
  ec: number;
  nitrogen: number;
  phosphorus: number;
  potassium: number;
  soil_moisture: number;
  soil_temperature: number;
  nutrient_status: Record<string, string>;
  recommendations: string[];
}

export interface SoilHistoryPoint {
  date: string;
  moisture: number;
  nitrogen: number;
  phosphorus: number;
  potassium: number;
  health_score: number;
  ec: number;
}

File: backend/app/schemas/soil.py
from pydantic import BaseModel
from datetime import datetime
from typing import Dict, Any

class SoilHealthReport(BaseModel):
    health_score: float
    degradation_risk: str
    organic_matter_percent: float
    ph: float
    ec: float
    nitrogen: float
    phosphorus: float
    potassium: float
    soil_moisture: float
    soil_temperature: float
    nutrient_status: Dict[str, str]
    recommendations: list[str]
    scientific_disclaimer: str = "Indicative field model estimate. Periodic lab calibration recommended."
    timestamp: datetime
